presto-analyzer/src/main/java/com/facebook/presto/sql/analyzer/MetadataResolver.java

```diff
@@ -87,4 +90,24 @@ default boolean isMaterializedView(QualifiedObjectName viewName)
     {
         return getMaterializedView(viewName).isPresent();
     }
+
+    /**
+     * Returns list of registered types.
+     */
+    List<Type> getTypes();
+
+    /**
+     * Returns list of registered parametric types.
+     */
+    List<ParametricType> getParametricTypes();
+
+    /**
+     * Returns list of registered functions.
```

Comment:
What's the difference between this method and the next? What does this method return that the next method doesn't? This should be in the JavaDoc. E.g. "Returns list of built-in and user defined functions" (though I don't know if that's accurate)